### ISSUE TITLE
Trilinos 12.18.1 and make 4.3 combination fails

### ISSUE THREAD

**Issue by MFraters** [dealii/candi#252]
When doing an install on Pop!_OS 21.04 (which is basically the same as Ubuntu), I notices that I got strange errors when candi was compiling Trilinos, or more specifically Kokkos, cause it to fail. Doing some research it looks like make 4.3 is breaking the compilation of Trilinos 12.18.1 (see for example https://github.com/GEOSX/thirdPartyLibs/issues/136). 

Proposed solutions are to downgrade make to 4.2 or upgrade Trilinos to 13.0.1. I tried upgrading Trilinos 13.0.1 with the following diff to `deal.II-toolchain/packages/trilinos.package`:
```
diff --git a/deal.II-toolchain/packages/trilinos.package b/deal.II-toolchain/packages/trilinos.package
index 1934e40..2823451 100644
--- a/deal.II-toolchain/packages/trilinos.package
+++ b/deal.II-toolchain/packages/trilinos.package
@@ -16,6 +16,15 @@ if [ ${TRILINOS_MAJOR_VERSION} = "DEV" ];then
     VERSION=dbf41f3b26b0326a1377f219e6e07eab861d181e
     EXTRACTSTO=trilinos-12.dbf41f
 
+    NAME=Trilinos.git
+    PACKING=git
+    SOURCE=https://github.com/trilinos/
+elif [ ${TRILINOS_MAJOR_VERSION} = "13" ];then
+    # download from github:
+    # this is the last "blessed" version we like:
+    VERSION=4796b92fb0644ba8c531dd9953e7a4878b05c62d
+    EXTRACTSTO=trilinos-13.0.1
+
     NAME=Trilinos.git
     PACKING=git
     SOURCE=https://github.com/trilinos/
@@ -184,7 +193,8 @@ CONFOPTS="\
   -D Trilinos_ENABLE_Ifpack2:BOOL=OFF \
   -D Trilinos_ENABLE_Tpetra:BOOL=ON \
   -D   Tpetra_INST_DOUBLE:BOOL=ON \
-  -D   Tpetra_INST_INT_LONG:BOOL=ON \
+  -D   Tpetra_INST_INT_LONG_LONG:BOOL=ON \
+  -D   Tpetra_INST_INT_LONG:BOOL=OFF \
   -D   Tpetra_INST_COMPLEX_DOUBLE:BOOL=ON \
   -D   Tpetra_INST_COMPLEX_FLOAT:BOOL=ON \
   -D   Tpetra_INST_FLOAT:BOOL=ON \
```

I needed the changes in INT_LONG's to pass the cmake checks. This compiles for me, but I haven't really tested it yet. Is using Trilinos 13 expected to be an issue for deal.ii? If so, I will downgrade my make next and just install Trilinos 12, but it might be good to add a warning/error when using using make 4.3.

**Comment by koecher** [comment]
@MFraters Thank you for reporting this issue with Trilinos 12 and make 4.3.

I recently used Trilinos 13 in an unmerged test branch, and yes, I can confirm the multiple integer issue.

We are currently considering if Trilinos 13 is supported by deal.II. For my experience, my codes worked as expected but there still can be other problems.

**Comment by tjhei** [comment]
addressed by #297
